Set toggl.sh executable by default

martinm76 opened the pull request:
Changed how to specify dates for ls and ical to be regular DATETIME objects
Updated help in toggl.py
Tested ical and ls with various permutations of DATETIME strings
Note: ical on purpose discards hours, minutes and seconds and sets start to 0:00:00 and end to 23:59:59

I hope this is in line with the existing date format?

If I can figure it out, I hope to add (display of) tag support next, if no one beats me to it.

martinm76 committed:
Set toggl.sh executable by default
Changed how to specify dates for ls and ical to be regular DATETIME objects
Updated help in toggl.py
Tested ical and ls with various permutations of DATETIME strings
Note: ical on purpose discards hours, minutes and seconds and sets start to 0:00:00 and end to 23:59:59

drobertadams commented:
I just tried this pull request and received the following. I was just trying to run toggl.py without any arguments.

Traceback (most recent call last):
  File "./toggl.py", line 1353, in <module>
    CLI().act()
  File "./toggl.py", line 1123, in act
    Logger.info(TimeEntryList())
  File "./toggl.py", line 61, in **call**
    cls.instance = super(Singleton, cls).**call**(_args, *_kw)
  File "./toggl.py", line 757, in **init**
    self.reload(start_time,stop_time)
  File "./toggl.py", line 813, in reload
    (TOGGL_URL, urllib.parse.quote(DateAndTime().parse_local_datetime_str(start_time).isoformat('T')), urllib.parse.quote(DateAndTime().parse_local_datetime_str(stop_time).isoformat('T')))
  File "./toggl.py", line 214, in parse_local_datetime_str
    return self.tz.localize( dateutil.parser.parse(datetime_str) )
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pytz/tzinfo.py", line 304, in localize
    raise ValueError('Not naive datetime (tzinfo is already set)')
ValueError: Not naive datetime (tzinfo is already set)

martinm76 committed:
o Fixed calling toggly.py without parameter errors
  o Added LOCATION = Client in IcalClientList
  o Added Listing of tags in description for IcalClientList and TimeEntryList

martinm76 commented:
Sorry about that. A new pull request is on its way where I fixed this and added a couple of other things.

martinm76 commented:
I hope my latest commit is showing up here now. I don't seem to be able to create another pull request at the moment. Should I close this one and create another, or do you get both commits here now?

drobertadams commented:
If you can close the original pull request yourself, perhaps you should do
that. But I'm not a github expert.

On Wed, Nov 11, 2015 at 3:19 AM martinm76 notifications@github.com wrote:

> I hope my latest commit is showing up here now. I don't seem to be able to
> create another pull request at the moment. Should I close this one and
> create another, or do you get both commits here now?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/drobertadams/toggl-cli/pull/51#issuecomment-155701206
> .
> 
> ## 
> 
> ## 
> 
> Dr. Robert Adams
> School of Computing and Information Systems
> Grand Valley State University

staff0rd commented:
You don't need to make a new pull request when fixing an issue specific to the current pull request.  All you do is just commit the fix to the branch the pull request was originally made from.  The pull request will always include all commits on that branch.  You can see this pull request currently has two commits attached to it.

@martinm76 to make your workflow simpler, you should create a separate branch when you make updates you intend to push back upstream.  When you're finished the branch, make the pull request from that branch.  It means you can keep contributing to your other branches whilst the pull request is in review.  If there's any conflicts, you can just do another commit to the pull request branch and it will automatically be included in the already raised pull request.

After the pull request is merged, you can then delete your pull request branch and merge upstream (which includes the pull request changes) into your master branch.

staff0rd commented:
@martinm76 After looking at your commits I might add that you are also including too many things in your commits/PRs, and your commit messages allude to this.  You should make pull requests that are specific to one feature or fix, as doing so makes the merges @drobertadams far simpler and avoids PRs getting rejected where they include both a desired change and an unrelated undesired change.

martinm76 commented:
I'll see about doing that, but this is the first thing I have contributed to on Github and both Github and Git are totally new to me as anything more than downloading a ZIP file from, and I have been quite confused by the terminology several times, so I was overjoyed that I could even commit anything.

I'll see if I can beef up my skills soon.
Might you be able to tell me how I commit to a certain branch, @staff0rd?

staff0rd commented:
@martinm76 no problem, this is the first project I've made pull requests to too.  Also the first time I've ever written python ha!  Unfortunately this also means I may also be doing it wrong ;)

Anyway, here's how I do it.  From your working directory you can then add this repo (drobertadams/toggl-cli) as the upstream remote like this;

```
$ git remote add upstream git@github.com:drobertadams/toggl-cli
$ git fetch upstream
```

Now whenever you want to make a change that you intend to later open a push request for, create a branch directly from the upstream remote;

```
$ git checkout -b my_new_fix upstream/master
```

You'll now be on the my_new_fix branch where you can commit until you feel the change is ready to be pushed, at which time you can push your branch to your fork.  The `-u` option sets the upstream of this branch to your fork;

```
$ git push -u origin my_new_fix
```

You can now use the github.com web UI to create a pull request from that branch.  If you need to make any changes to it, at any time before your PR is merged you can make more commits and just call `git push` and those commits will appear in the PR.

This workflow allows you to create further feature/fix branches for the same project while your PR is still open, just be creating new branches and switching between them as needed.

martinm76 commented:
Thank you, Stafford!. I'll refer back to this set of instructions very soon
;)

fre. 13. nov. 2015 kl. 06.47 skrev Stafford Williams <
notifications@github.com>:

> @martinm76 https://github.com/martinm76 no problem, this is the first
> project I've made pull requests to too. Also the first time I've ever
> written python ha! Unfortunately this also means I may also be doing it
> wrong ;)
> 
> Anyway, here's how I do it. From your working directory you can then add
> this repo (drobertadams/toggl-cli) as the upstream remote like this;
> 
> $ git remote add upstream git@github.com:drobertadams/toggl-cli
> $ git fetch upstream
> 
> Now whenever you want to make a change that you intend to later open a
> push request for, create a branch directly from the upstream remote;
> 
> $ git checkout -b my_new_fix upstream/master
> 
> You'll now be on the my_new_fix branch where you can commit until you feel
> the change is ready to be pushed, at which time you can push your branch to
> your fork. The -u option sets the upstream of this branch to your fork;
> 
> $ git push -u origin my_new_fix
> 
> You can now use the github.com web UI to create a pull request from that
> branch. If you need to make any changes to it, at any time before your PR
> is merged you can make more commits and just call git push and those
> commits will appear in the PR.
> 
> This workflow allows you to create further feature/fix branches for the
> same project while your PR is still open, just be creating new branches and
> switching between them as needed.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/drobertadams/toggl-cli/pull/51#issuecomment-156329204
> .
